web3_proxy/src/frontend/mod.rs

```diff
@@ -229,9 +235,20 @@ pub async fn serve(port: u16, proxy_app: Arc<Web3ProxyApp>) -> anyhow::Result<()
     let service = app.into_make_service_with_connect_info::<SocketAddr>();
 
     // `axum::Server` is a re-export of `hyper::Server`
-    axum::Server::bind(&addr)
+    let server = axum::Server::bind(&addr)
         // TODO: option to use with_connect_info. we want it in dev, but not when running behind a proxy, but not
         .serve(service)
+        // <<<<<<< HEAD
+        //         .with_graceful_shutdown(async move {
+        //             let _ = shutdown_receiver.recv().await;
+        //         })
+        //         .await
+        //         .map_err(Into::into);
+        //
+        //     let _ = shutdown_complete_sender.send(());
+        //
+        //     server
+        // =======
```

Comment:
i'm pretty sure we still need this. otherwise I think the app is going to kill active frontend requests

Reply:
on the main branch this seems to be completely removed

Reply:
stats_v2 has this: 
```
    // `axum::Server` is a re-export of `hyper::Server`
    let server = axum::Server::bind(&addr)
        // TODO: option to use with_connect_info. we want it in dev, but not when running behind a proxy, but not
        .serve(service)
        .with_graceful_shutdown(async move {
            let _ = shutdown_receiver.recv().await;
        })
        .await
        .map_err(Into::into);

    let _ = shutdown_complete_sender.send(());

    server
    ```
    So I will take it over from there

Reply:
resolving this because that seems to be the way forward